tools/harness-thci/OpenThread.py

```diff
@@ -2963,13 +2963,19 @@ def setBbrDataset(self, SeqNumInc=False, SeqNum=None, MlrTimeout=None, ReRegDela
             False: fail to set BBR Dataset
         """
         assert not (SeqNumInc and SeqNum is not None), "Must not specify both SeqNumInc and SeqNum"
+
+        if MlrTimeout and MlrTimeout != self.bbrMlrTimeout or ReRegDelay and ReRegDelay != self.bbrReRegDelay:
+            SeqNumInc = True
```

Comment:
Instead of setting `SeqNumInc = True`, should we add `assert` to make sure Harness is not passing incorrect parameters?

```suggestion
            assert(SeqNumInc == True)
```

@karthick-grl @edmont

Reply:
This is the reason why the change is needed, some test scripts pass a single parameter like `MlrTimeout` or `ReRegDelay`, but not `SeqNumInc`, while according to the specification the sequence number should be increased with each change of the BBR Dataset.

I believe there are different ways or places where to introduce this logic:

- Test scripts.
- THCI.
- Device firmware.

Currently the safest method seems to be the THCI. Moreover, Kirale's THCI is already doing something similar.

Reply:
SGTM.